packages/cli/src/scripts/validate-features.ts

```diff
@@ -0,0 +1,244 @@
+      --verbose, -v Enable verbose output
+      --fail-early, -e Exit on first failure
+
+    Examples
+      $ validate-features -e
+`,
+  {
+    flags: {
+      failEarly: {
+        type: 'boolean',
+        alias: 'e',
+      },
+      verbose: {
+        type: 'boolean',
+        alias: 'v',
+      },
+    },
+  }
+);
+
+type Failure = {
+  feature: Feature;
+  messages: string[];
+};
+
+async function main() {
+  const directory = defaultJsonDirectory;
+  const isVerbose = cli.flags.verbose;
+
+  console.log('Fail early?', cli.flags.failEarly);
+
+  const schemaInfo = getSchemaInfo(directory);
+
+  const allFailures: Failure[] = [];
+
+  /**
+   * The features configuration is imported dynamically here. We could include
+   * it in the common bundle and import from there, but it feels a little
+   * annoying having to place an app configuration in the common bundle and
+   * build it before it becomes effective.
+   */
+  const { features } = (await import(featureFlagsConfigFile)) as {
+    features: Feature[];
+  };
+
+  for (const feature of features) {
+    if (isVerbose) {
+      console.log(
+        `Feature ${feature.name}: ${feature.isEnabled ? 'enabled' : 'disabled'}`
+      );
+    }
+
+    const messages = await validateFeatureData(feature, schemaInfo);
+
+    if (messages) {
+      allFailures.push({ feature, messages });
+    }
+
+    if (cli.flags.failEarly && !isEmpty(allFailures)) {
+      console.log(
+        'Found failures with --fail-early enabled, so skipping the rest...'
+      );
+      break;
+    }
+  }
+
+  if (!isEmpty(allFailures)) {
+    allFailures.forEach((failure) =>
+      logError(
+        `Failed to validate ${
+          failure.feature.isEnabled ? 'enabled' : 'disabled'
+        } feature ${failure.feature.name}:\n${failure.messages}`
+      )
+    );
+    logError(
+      `There were ${allFailures.length} features that failed to validate their data.`
+    );
+    process.exit(1);
+  } else {
+    logSuccess('All features are validated!');
+  }
+}
+
+main().then(
+  () => process.exit(0),
+  (err: Error) => {
+    logError(err.message);
+    process.exit(1);
+  }
+);
+
+/**
+ * Validate if the data is there or not there. Returns one validation error
+ * message per scope.
+ */
+async function validateFeatureData(feature: Feature, schemaInfo: SchemaInfo) {
+  /**
+   * First we check if there are metric properties involved, because in that
+   * case we only check the properties and do not enforce existence/absence of
+   * the full metric.
+   */
+  if (feature.metricProperties) {
+    assert(
+      feature.metricScopes,
+      `Missing metricScopes configuration for feature ${feature.name}`
+    );
+
+    assert(
+      feature.metricName,
+      'If a feature defines metricProperties it should also have a metricName'
+    );
+
+    const promisedResults = feature.metricScopes.map((scope) =>
+      validateMetricPropertiesForScope(
+        feature.metricName!,
+        feature.metricProperties!,
```

Comment:
I think you can get rid of the `!` when yuo assign the properties to variables
```suggestion
    const { metricScopes, metricName } = feature;
    
    assert(
      metricScopes,
      `Missing metricScopes configuration for feature ${feature.name}`
    );

    assert(
      metricName,
      'If a feature defines metricProperties it should also have a metricName'
    );

    const promisedResults = feature.metricScopes.map((scope) =>
      validateMetricPropertiesForScope(
        metricName,
        metricProperties,
```

Reply:
Yes, that works 👍